internal/internal_workflow_client.go

```diff
@@ -1699,7 +1695,12 @@ func (uh *updateHandle) UpdateID() string {
 }
 
 func (uh *updateHandle) Get(ctx context.Context, valuePtr interface{}) error {
-	if uh.err != nil {
+	// implementation note: this is a broken implementation that assumes the
+	// update has already completed by the time Get is called. This is true for
+	// the current implementation of synchronous updates but will not be true in
+	// the future with async update.
+
+	if uh.err != nil || valuePtr == nil {
```

Comment:
To confirm, what you're saying here is that you acknowledge that in the future when I pass `nil` for `valuePtr`, you'll change this code to properly wait on the update? (what the `valuePtr` value is only applies _after_ completion)

Reply:
Yes. The semantics should be exactly the same as Future.Get to make things as unsurprising as possible.

Reply:
:+1: Will try to set a reminder for when non-sync update comes around